mlir/utils/jenkins/MIOpenDriver.py

```diff
@@ -0,0 +1,479 @@
+        # print Wi
+        result = result + str(self.wi) + ','
+        # print K
+        result = result + str(self.k) + ','
+        # print Y
+        result = result + str(self.y) + ','
+        # print X
+        result = result + str(self.x) + ','
+        # print dilation height
+        result = result + str(self.dilationH) + ','
+        # print dilation width
+        result = result + str(self.dilationW) + ','
+        # print stride height
+        result = result + str(self.convStrideH) + ','
+        # print stride width
+        result = result + str(self.convStrideW) + ','
+        # print padding width
+        result = result + str(self.paddingH) + ','
+        # print padding height
+        result = result + str(self.paddingW) + ','
+
+        # benchmarking fields
+
+        # print TFlops
+        result = result + str(self.computeTFlops(nanoSeconds))
+        return result
+
+    def generateMlirDriverCommandLine(self):
+        result = ''
+        # set operation.
+        if self.direction == 'fwd':
+            result = result + '--operation conv2d'
+        elif self.direction == 'bwd':
+            result = result + '--operation conv2d_bwd_data'
+        elif self.direction == 'wrw':
+            result = result + '--operation conv2d_bwd_weight'
+        # set data type.
+        result = result + ' -t ' + self.dataType
+        # set XDLOPS
+        if self.xdlops == True:
+            result = result + ' -x2'
+        # set filter layout.
+        result = result + ' --fil_layout ' + self.filterLayout
+        # set input layout.
+        result = result + ' --in_layout ' + self.inputLayout
+        # set output layout.
+        result = result + ' --out_layout ' + self.outputLayout
+        # set N
+        result = result + ' --batchsize ' + str(self.n)
+        # set C
+        result = result + ' --in_channels ' + str(self.c)
+        # set Hi
+        result = result + ' --in_h ' + str(self.hi)
+        # set Wi
+        result = result + ' --in_w ' + str(self.wi)
+        # set K
+        result = result + ' --out_channels ' + str(self.k)
+        # set Y
+        result = result + ' --fil_w ' + str(self.y)
+        # set X
+        result = result + ' --fil_h ' + str(self.x)
+        # set dilation height
+        result = result + ' --dilation_h ' + str(self.dilationH)
+        # set dilation width
+        result = result + ' --dilation_w ' + str(self.dilationW)
+        # set stride height
+        result = result + ' --conv_stride_h ' + str(self.convStrideH)
+        # set stride width
+        result = result + ' --conv_stride_w ' + str(self.convStrideW)
+        # set padding width
+        result = result + ' --padding_h ' + str(self.paddingH)
+        # set padding height
+        result = result + ' --padding_w ' + str(self.paddingW)
+        return result
+
+    def __init__(self, argv, xdlops):
+        # setup default values.
+        self.dataType = 'f32'
```

Comment:
From a defensiveness point of view, wouldn't it be better not to assign default values to most of these parameters and instead error if they're not all present. Last I checked, these settings are always passed in - having them silently default is a recipe for errors.